src/proxy.rs

```diff
@@ -107,9 +108,39 @@ pub trait IntoProxyScheme {
     fn into_proxy_scheme(self) -> crate::Result<ProxyScheme>;
 }
 
-impl<T: IntoUrl> IntoProxyScheme for T {
+impl<S: AsRef<str>> IntoProxyScheme for S {
     fn into_proxy_scheme(self) -> crate::Result<ProxyScheme> {
-        ProxyScheme::parse(self.into_url()?)
+        let mut url = match Url::parse(self.as_ref()) {
```

Comment:
Is there a reason to go straight to `Url::parse`, instead of just calling `into_url` first?

Reply:
Yeah, there is. I'm trying to remember the exact problem, if I go to `into_url` then I can't check the scheme, some other problem occurs. I'll try what you suggest to see if I can better explain what the problem is.

Reply:
Oops, you're right of course. I was confused by some incorrect approaches I had taken previously. I've now pushed the simpler version [37343c6](https://github.com/seanmonstar/reqwest/compare/308c6793f146f85a9b369cf3da610ad42c4154bb..37343c67d11afc65d4a83510984deab93f93fe5c)